server/controllers/comment.js

```diff
@@ -117,6 +120,35 @@ class Comment {
 
     return goodHttpResponse(response, 200, 'Comment and edit history found', comment);
   }
+
+  /**
+   * Get comments with the replies
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Comment Object
+   */
+  static async getComments(request, response) {
+    const { slug } = request.params;
+
+    const article = await articleRepo.getArticleBySlug(slug);
+    if (article === null) {
+      return badHttpResponse(response, 404, 'We could not find this article');
+    }
+
+    const rawComments = await commentRepo.getCommentsOnArticle(article.id);
+    const maxParentId = Math.max(...rawComments.map(element => element.parentId));
```

Comment:
Why are we spreading the array in the max method? We could simple map through and then pass the array to the Math.max function

Reply:
Math.max takes numeric parameters. The spread is required in order to parse the parentId as parameters to the Math.max method.